p2p/host/basic/basic_host.go

```diff
@@ -87,8 +87,14 @@ type BasicHost struct {
 	mx        sync.Mutex
 	lastAddrs []ma.Multiaddr
 	emitters  struct {
+		evtPeerStateChange       event.Emitter
 		evtLocalProtocolsUpdated event.Emitter
 	}
+
+	stripedPeerConnectivity [256]struct {
+		sync.Mutex
+		observedConnectedness map[peer.ID]network.Connectedness
+	}
```

Comment:
Can we push this state down into the _notifee => eventbus shim_? It doesn't really belong here. IUC, it's only used to enforce non-idempotent firing.